src/frontend/screens/Observation/Buttons.tsx

```diff
@@ -104,11 +104,16 @@ export const ButtonFields = ({
   async function handlePressShare() {
     const {lon, lat} = observation;
     setShareButtonLoading(true);
-
-    const urlsQueries = await Promise.all(
-      attachmentUrlQueries.map(q => q.refetch()),
-    );
-    const urls = urlsQueries.map(query => query.data!.url);
+    const needsRefetch = attachmentUrlQueries.some(query => !query.data);
+    let urls;
+    if (needsRefetch) {
+      const urlsQueries = await Promise.all(
+        attachmentUrlQueries.map(q => q.refetch()),
+      );
+      urls = urlsQueries.map(query => query.data!.url);
+    } else {
+      urls = attachmentUrlQueries.map(query => query.data!.url);
+    }
```

Comment:
i would be careful about using the `!` operator. that's essentially similar to telling TypeScript that we know that the value is guaranteed to not be null/undefined, which is something that we can't guarantee based on the code here.

Reply:
Do you think the refetch is useful to keep here? I'm not sure it's necessary to keep, as it seemed to be what was causing the original issue before.

Potential suggestion:

- Don't refetch. Live with the current state of the attachement url queries whenever the button is pressed
- if any of the queries are missing `data`, either:
  - surface an error in the UI that prevents sharing from happening as a whole
  - filter out the attachments where `data` is not present and continue to process the ones that we have a url for
  
Not sure which of these are preferred from a UX perspective, but this additional handling should be considered anyways because query-refetching doesn't guarantee that a valid url will be retrieved

Reply:
I eliminated the ! operator.
I have to refetch. Without that refetch there is never any data in the attachments when you first go to share. That is why I kept it in and made the if else. If there is another way to get that data in there the first time, I could not find it and would love to hear your ideas. Sorry. I think that is why software mansion had it in there.
I applied the filter as you suggested where the data was not present and throw an alert error if there is no data present. Not sure if that is preferred or if we should just share without the photos?

Reply:
> I have to refetch. Without that refetch there is never any data in the attachments when you first go to share. That is why I kept it in and made the if else. If there is another way to get that data in there the first time, I could not find it and would love to hear your ideas. Sorry. I think that is why software mansion had it in there.

ah i see. yeah not sure what the issue is without investigating a bit. hard to tell at a glance.

> I applied the filter as you suggested where the data was not present and throw an alert error if there is no data present. Not sure if that is preferred or if we should just share without the photos?

yeah i'm not sure what's preferred. think what you have makes sense and is easy to change later if needed.

Reply:
> I have to refetch. Without that refetch there is never any data in the attachments when you first go to share.

figured out the issue: 

https://github.com/digidem/comapeo-mobile/blob/0514c3b77647d493beba05c31ca39394ccb707f3/src/frontend/screens/Observation/Buttons.tsx#L84

should be omitted or `true` instead in order for the query to run.

Reply:
> I applied the filter as you suggested where the data was not present and throw an alert error if there is no data present. Not sure if that is preferred or if we should just share without the photos?

going back to this, i guess the safest approach (for now) would be to do whatever mapeo mobile does or something similar? based on:

https://github.com/digidem/mapeo-mobile/blob/develop/src/frontend/screens/Observation/ObservationView.js#L152

seems like it leans towards the all-or-nothing approach. what you currently have seems most aligned with that, so i think that's fine